Base/QTGUI/qSlicerApplication.cxx

```diff
@@ -396,21 +396,24 @@ bool qSlicerApplication::notify(QObject *receiver, QEvent *event)
   catch( std::exception& exception )
     {
     QString errorMessage;
-    errorMessage = tr("%1 has caught an internal error.\n\n").arg(this->applicationName());
-    errorMessage += tr("You may be able to continue from this point, but results are undefined.\n\n");
-    errorMessage += tr("Suggested action is to save your work and restart.");
-    errorMessage += tr("\n\nIf you have a repeatable sequence of steps that causes this message, ");
+    errorMessage = tr("%1 has caught an exception, ").arg(this->applicationName());
+    errorMessage += tr("please save your work and restart.\n\n");
+    errorMessage += tr("If you have a repeatable sequence of steps that causes this message, ");
     errorMessage += tr("please report the issue following instructions available at http://slicer.org\n\n\n");
     errorMessage += tr("The message detail is:\n\n");
     errorMessage += tr("Exception thrown in event: ") + exception.what();
+    if (!tr("bad allocation").compare(exception.what()) || !tr("std::bad_alloc").compare(exception.what()))
```

Comment:
Here it would be better to compare the exception type directly, rather than the exception.what() string, which may be different on depending on the platform (especially it may not match the localized string from tr).

https://en.cppreference.com/w/cpp/memory/new/bad_alloc

Reply:
I changed the method of checking exception types. The new method should be able to capture the target exception type robustly.

I did not find a good way to compare the exception type directly. The cppreference uses std::strcmp(what(), other.what()) for comparing exception types in all the exception pages. I avoided using cstring as an additional lib. Please let me know if there are any good resources on such type comparison topic.